### ISSUE TITLE
re-design support for skip and refresh events in ASM

### ISSUE THREAD

**Issue by kiniry** [FreeAndFair/ColoradoRLA#239]
Deal with this comment on PR #235.

```server/eclipse-project/src/main/java/us/freeandfair/corla/asm/AbstractStateMachine.java
@@ -217,6 +228,16 @@ public ASMState stepTransition(final ASMTransition the_transition)
    */
   public ASMState stepEvent(final ASMEvent the_event)
       throws IllegalStateException {
+    // refresh and skip events are always permitted
```
      @dmzimmerman

Eek. I don't like having these in the AbstractStateMachine class directly. Is there a way that we can annotate the events in the enumeration instead (like a method isAlwaysEnabled()), or throw them in a collection of always-enabled events that can be added to by child classes, or something?

**Comment by kiniry** [comment]
@dmzimmerman is revising the server ASM now in #327.

**Comment by dmzimmerman** [comment]
This is addressed in #327 by virtual of skip and refresh events no longer existing.
